test/e2e/networkingnew.go

```diff
@@ -0,0 +1,358 @@
+	}
+	sem := make(chan string, len(ports))
+	for p := range ports {
+
+		go func(nodes *api.NodeList, port int) {
+
+			var svcname = fmt.Sprintf("nettest-%v", port)
+
+			By(fmt.Sprintf("creating a service named %q in namespace %q", svcname, f.Namespace.Name))
+			svc, err := f.Client.Services(f.Namespace.Name).Create(&api.Service{
+				ObjectMeta: api.ObjectMeta{
+					Name: svcname,
+					Labels: map[string]string{
+						"name": svcname,
+					},
+				},
+				Spec: api.ServiceSpec{
+					Ports: []api.ServicePort{{
+						Protocol:   "TCP",
+						Port:       port,
+						TargetPort: util.NewIntOrStringFromInt(port),
+					}},
+					Selector: map[string]string{
+						"name": svcname,
+					},
+				},
+			})
+
+			if err != nil {
+				Failf("unable to create test service named [%s] %v on port %v", svc.Name, err, port)
+			} else {
+				Logf("************************** Created service successfully [%s]", svc.Name)
+			}
+
+			// Clean up service
+			defer func() {
+				By("Cleaning up the service")
+				if err := f.Client.Services(f.Namespace.Name).Delete(svc.Name); err != nil {
+					Failf("unable to delete svc %v: %v", svc.Name, err)
+				}
+			}()
+
+			Logf("launching pod per node.....")
+			podNames := launchNetTestPodPerNode(port, nettestVersion, f, nodes, svcname)
+			// Clean up the pods
+			defer func() {
+				By("Cleaning up the webserver pods")
+				for _, podName := range podNames {
+					if err = f.Client.Pods(f.Namespace.Name).Delete(podName, nil); err != nil {
+						Logf("Failed to delete pod %s: %v", podName, err)
+					}
+				}
+			}()
+
+			Logf("************************** Launched test pods for %v", port)
+			By("waiting for all of the webserver pods to transition to Running + reaching the Passing state.")
+
+			for _, podName := range podNames {
+				err = f.WaitForPodRunning(podName)
+				Expect(err).NotTo(HaveOccurred())
+				By(fmt.Sprintf("waiting for connectivity to be verified [ port =  %v ] ", port))
+				//once response OK, evaluate response body for pass/fail.
+				pollPeerStatus(sem, f, svc, timeoutSeconds)
+			}
+
+			Logf("************************** Finished test pods for %v", port)
+		}(nodes, ports[p])
+	}
+	//now wait for the all X nettests to complete...
+	for pReturned := range ports {
+		Logf("Waiting on ports to report back.  So far %v have been discovered...", pReturned)
+		Logf("... Another service has successfully been discovered: %v ( %v ) ", pReturned, <-sem)
+	}
+	Logf("Completed test on %v port/services", len(ports))
+}
+
+//makePorts makes a bunch of ports from 8080->8080+n
+func makePorts(n int) []int {
+	m := make([]int, n)
+	for i := 0; i < n; i++ {
+		m[i] = 8080 + i
+	}
+	return m
+}
+
+//launchNetTestPodPerNode launches nettest pods, and returns their names.
+func launchNetTestPodPerNode(port int, version string, f *Framework, nodes *api.NodeList, name string) []string {
+	podNames := []string{}
+
+	totalPods := len(nodes.Items)
+
+	Expect(totalPods).NotTo(Equal(0))
```

Comment:
Yup, its just  a necessary precondition, but not necessary in this test.    This function indeed is one which can be called from different places...  I'll leave this asis for now.